Cleanup `keymap_support.c`

SiriusStarr opened the pull request:
Some of this is formatting-related.  Do we have a standard formatter to run?  And if not, should we?

If the automouse flags are just going to always be set, then we should probably remove all the (non-console/debug) ifdefs?  My concern is ending up with some odd combination of flags such that things don't compile because we're accessing something not being defined or whatnot.

Anyways, this all builds and works fine for me, but more eyes on it would be good, obviously.

SiriusStarr committed:
Remove always enabled ifdefs around Sval features

Things like Achordion shouldn't be dependent on pointing devices anyways.

SiriusStarr committed:
Cleanup some inconsistent whitespace/indentation

SiriusStarr committed:
Restructure custom key handling

This moves all non-hold keys to activate on key-down, so we only need to
think about clearing held states in the key-up case.  It also aborts
further handling of these keycodes once we're done with them.

This fixes the issue with caps word toggle not actually toggling and
getting processed multiple times.  It does prevent users from hooking these
custom keys, but if people know how to do that, they can probably figure
out how to make their own custom key that calls the same code.

SiriusStarr committed:
Remove duplicate PS2_MOUSE_SCROLL_BTN_MASK

This is already defined in `trackpoint/config.h`. If you have input @ilc I'm all ears.

ilc commented on keyboards/svalboard/keymaps/keymap_support.c:
```diff
@@ -256,110 +252,109 @@ bool process_record_kb(uint16_t keycode, keyrecord_t *record) {
     }
-#endif
 
     return process_record_user(keycode, record);
```
While we are here, this should goto the top of the function, so it actually respects user stuff over svalboard.

SiriusStarr replied:
Good catch; resolved per QMK standards.

ilc reviewed (changes requested):
I still need to pull this locally for testing.  Assuming it's good and the change requested is good.  We'll be good to go.

SiriusStarr committed:
Clarify debug message

SiriusStarr committed:
Fix skipping matrix_scan_user with automouse inf

SiriusStarr committed:
Run userspace record handling before keyboard

This is QMK convention, per https://docs.qmk.fm/hardware_keyboard_guidelines#custom-keyboard-programming

ilc commented:
LGTM - Running on the board I am using right now.
